lib/admin_constraint.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require_dependency 'current_user'
+
+class AdminConstraint
```

Comment:
@castrolem I'm not sure if this works, there are references to a Discourse class in the method. The idea is to adapt this code so it works with our `current_user` implementation.